### PR TITLE
Use file extension for CSRG format detection

### PR DESCRIPTION
**Pull request by NebelNidas**
~~If the passed path is a file and has an extension, use this shortcut instead of parsing file contents~~

Now we can detect CSRG at least via the path-based API

### CODE REVIEW

File: src/main/java/net/fabricmc/mappingio/MappingReader.java [hunk 1]
```diff
@@ -43,6 +43,26 @@ public static MappingFormat detectFormat(Path file) throws IOException {
...
+			String fileName = file.getFileName().toString();
+			int dotIdx = fileName.lastIndexOf('.');
+
+			if (dotIdx >= 0) {
+				String ext = fileName.substring(dotIdx + 1);
+
+				switch (ext) {
+				case "mapping": // Standard name
+				case "mappings": // Sometimes used for standalone (non-dir) files
+				case "enigma": // Erroneously used by Lorenz <0.6
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/mappingio/MappingReader.java:55]: Why support it then?

File: src/main/java/net/fabricmc/mappingio/MappingReader.java [hunk 2]
```diff
@@ -43,6 +43,26 @@ public static MappingFormat detectFormat(Path file) throws IOException {
...
 		} else {
+			String fileName = file.getFileName().toString();
+			int dotIdx = fileName.lastIndexOf('.');
+
+			if (dotIdx >= 0) {
+				String ext = fileName.substring(dotIdx + 1);
+
+				switch (ext) {
+				case "mapping": // Standard name
+				case "mappings": // Sometimes used for standalone (non-dir) files
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/mappingio/MappingReader.java:54]: Im not convinved this is a good idea for more general names like mapping.

File: src/main/java/net/fabricmc/mappingio/MappingReader.java (hunk 1 above)
Comment (NebelNidas) [reply src/main/java/net/fabricmc/mappingio/MappingReader.java:55]: Why not? It's unique enough to not cause any issues

Comment (modmuss50) [reply src/main/java/net/fabricmc/mappingio/MappingReader.java:55]: Maybe add an overload to opt out of this ext detection?

Comment (NebelNidas) [reply src/main/java/net/fabricmc/mappingio/MappingReader.java:57]: Done

File: src/main/java/net/fabricmc/mappingio/MappingReader.java (hunk 2 above)
Comment (NebelNidas) [reply src/main/java/net/fabricmc/mappingio/MappingReader.java:56]: Solved by adding the `trustFileExtension` parameter

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**